cf-custom-resources/lib/wkld-custom-domain.js

```diff
@@ -185,13 +185,19 @@ exports.handler = async function (event, context) {
     switch (event.RequestType) {
       case "Update":
         let oldAliases = new Set(event.OldResourceProperties.Aliases);
-        if (setEqual(oldAliases, aliases)) {
+        let oldHostedZoneId = event.OldResourceProperties.PublicAccessHostedZoneID;
+        let oldDNS = event.OldResourceProperties.PublicAccessDNS;
+        if (setEqual(oldAliases, aliases) && oldHostedZoneId === publicAccessHostedZoneID && oldDNS === publicAccessDNS) {
```

Comment:
Maybe we need proper comment to explain why we do this now (why we check `publicAccessHostedZoneID` and `publicAccessDNS`).

Reply:
I feel like the assumption that "hostedzone and dns never changed" was quite a tight assumption previously, and it was super implementation specific..... Adding these conditions actually make a lot of sense, and would totally make sense if we add the same thing to the other custom domain codes as well...

But agree that a comment would be even nicer!